models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -350,17 +359,13 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
             }
         }
 
-        if ($contentOptimizedFormat && !self::hasWebpSupport()) {
-            $format = $image->getContentOptimizedFormat();
```

Comment:
Why have you removed that section? 
As I said: `getContentOptimizedFormat()` is a completely different functionality, it returns the image type (PNG or JPEG) depending on the image content, it has nothin to do with any optimization.

Reply:
It hijacks the format because it has no regard for the actually requested format.
So if you've got a png without alpha and format is `png`, the function will set the format to `pjpeg` - regardless if some sort of "auto-detection" is requested.
As you said, it's a completely different function and jet it still is/was somehow related to the optimization pipeline (?)
However, restored the code.

Reply:
That whole thing bothered me to much. I've split the flag indicating content and / or format optimization into two separate variables. Format optimization is only allowed if no specific format is requested.